### PR TITLE
[hareflow] Minor update + add vcpkg_cmake_config_fixup

### PR DESCRIPTION
**Pull request by klalumiere**
# Changes

- Add `vcpkg_cmake_config_fixup()`. Without it, we had problems like
    ```shell
    CMake Warning at $VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake:639 (_add_library):
      Cannot generate a safe dependent library path for target CoveoTarget because
      there is a cycle in the constraint graph:

        dir 0 is [/builds/build/vcpkg_installed/arm64-linux-dynamic/debug/lib]
          dir 1 must precede it due to runtime library [libhareflow.so]
        dir 1 is [/builds/build/vcpkg_installed/arm64-linux-dynamic/lib]
          dir 0 must precede it due to runtime library [libaws-cpp-sdk-sqs.so]

      Some of these libraries may not be found correctly.
    Call Stack (most recent call first):
      CoveoTarget/CMakeLists.txt:1 (add_library)
    ```
    because `IMPORTED_LOCATION_DEBUG` was missing.
- Minor update to the port from version 0.1.0 to 0.1.1. This integrates a pull request by @BillyONeal (thanks!) where some missing headers were added. Following this pull request, we don't need the patch anymore.

c.c. @yrivardmulrooney, @dalloriam: this might interest you.

# Housekeeping

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/hareflow/portfile.cmake [hunk 1]
```diff
@@ -21,5 +19,7 @@ vcpkg_cmake_configure(
 )
 vcpkg_cmake_install()
 
+vcpkg_cmake_config_fixup()
+
 file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (LilyWangLL) [inline ports/hareflow/portfile.cmake:24]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

**Comment by LilyWangLL** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/hareflow/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "hareflow",
-  "version-semver": "0.1.0",
-  "port-version": 1,
+  "version-semver": "0.1.1",
```
[the remark this answers is not in the document]
Comment (klalumiere) [reply ports/hareflow/vcpkg.json:3]: Thanks for the review!

Yes, this is a semver library. 🙂

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**